tests/extensions/test_eo.py

```diff
@@ -441,3 +442,33 @@ def test_older_extension_version(ext_item: Item) -> None:
     migrated_item = pystac.Item.from_dict(item_as_dict, migrate=True)
     assert EOExtension.has_extension(migrated_item)
     assert new in migrated_item.stac_extensions
+
+
+@pytest.mark.parametrize(
+    "filter,count",
+    [
+        (dict(), 11),
+        ({"name": "B4"}, 1),
+        ({"common_name": "blue"}, 1),
+        ({"name": "B4", "common_name": "red"}, 1),
+        ({"name": "B4", "common_name": "green"}, 0),
+    ],
+)
+def test_get_assets(ext_item: pystac.Item, filter: Dict[str, str], count: int) -> None:
+    assets = EOExtension.ext(ext_item).get_assets(**filter)  # type:ignore
```

Comment:
The need to use `# type: ignore` here is a design smell, not with this PR, but with how extensions are implemented. Just noting that here so that we might remember later to design an extension interface that doesn't need this :-). No action required on your part.

Reply:
Yeah totally. I messed around a bit with changing the output type of `EOExtension.ext()` to make it a `Union[ItemEOExtension, AssetEOExtension]` but then I didn't feel comfortable making that change for only this extension :(

Reply:
Yeah, returning unions generally feels bad to me (unless it's `Union[None, Thing]`) so I'm ok with how things are, thanks.